### DISCUSSION TITLE
Pivoting data from InfluxDB puts metadata into column names.

### DISCUSSION THREAD

**Discussion by Balta-Stefan** [grafana/grafana#53871] [Dashboards, Panels & Visualizations]
**What happened**:
After performing pivoting with the following query:

```   
      from(bucket: "bucket")    
      |> range(start: -3h)    
      |> filter(fn: (r) => r["_measurement"] == "city aggregation")    
      |> filter(fn: (r) => r["_field"] == "longitude")    
      |> pivot(    
          rowKey:["_time"],    
          columnKey: ["_field"],    
          valueColumn: "_value"    
  )
```

I get the following column names:

![alt text](https://i.imgur.com/FcmFfuG.png)

In order to change the name of the column with curly brackets, I have to use rename by regex transformation.

**What you expected to happen**:

I expected the column names to simply be "_time" and "longitude".

**Environment**:
- Grafana version: v9.0.3
- Data source type & version: InfluxDB v2.3.0+SNAPSHOT.090f681737
- OS Grafana is installed on: Docker

**Comment by zuchka** [comment]
Thanks for creating this issue, @Balta-Stefan  

Here is a nice Influx2 dashboard on our public sandbox: https://play.grafana.org/d/2N0n0Er7k/influx-2-2-live-noaa-buoy-data?orgId=1

do you think you could try and replicate your issue on this instance? :pray:

**Comment by Balta-Stefan** [comment]
> Thanks for creating this issue, @Balta-Stefan
> 
> Here is a nice Influx2 dashboard on our public sandbox: https://play.grafana.org/d/2N0n0Er7k/influx-2-2-live-noaa-buoy-data?orgId=1
> 
> do you think you could try and replicate your issue on this instance? 🙏

I didn't even have to do any queries, I just hovered over a point on the existing Geomap and the same issue happened.
![Problem](https://user-images.githubusercontent.com/58751753/183631784-b6a9cd60-3286-435b-b909-25eb5f9f4cf6.png)

Basically, every column contains the entire table in its name, in JSON format.

EDIT: I can see that [someone else](https://community.grafana.com/t/meta-data-in-column-name-with-grafana-influxdb-table-panel/52169/5) had the same problem.

**Comment by RobbertH** [comment]
I have the same issue even without pivoting. I'm using a similar setup (Grafana + InfluxDBv2 in docker).
Do you think it could have something to do with the fact that I'm using the Flux query language, which is in beta?

**Comment by RobbertH** [comment]
Might've found something that helps: in the "transform" tab, you can add a "Labels to fields" transformation.

**Reply by leandrolanzieri** [reply]
this fixed my issue!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
